src/app/shared/change-password-dialog/change-password-dialog.component.html

```diff
@@ -0,0 +1,47 @@
+<h1 mat-dialog-title>Change Password</h1>
+
+<div mat-dialog-content>
+
+  <form [formGroup]="changePasswordForm">
+
+    <div fxLayout="column">
+
+      <mat-form-field>
+        <mat-label>Password</mat-label>
+        <input matInput formControlName="password" type="password">
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('required')">
+          Password <strong>is required</strong>
+        </mat-error>
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('minlength')">
+          Password should be at least <strong>8 characters long</strong>
+        </mat-error>
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('maxlength')">
+          Password should not be more than <strong>50 characters long</strong>
+        </mat-error>
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('pattern')">
+          Pattern does not match
```

Comment:
change error to "password should include a **numeral** and at **least one uppercase, lowercase and special character**"

Reply:
Can I give the maximum and minimum number of characters with in the pattern by replacing Validators.maxLength and Validators.minLength ?

Reply:
no its fine